### ISSUE TITLE
Something is wrong when the inherit value is missing.

### ISSUE THREAD

**Issue by HansBug** [opendilab/treevalue#10]
- [x] I have marked all applicable categories:
    + [ ] installation bug
    + [ ] exception-raising bug
    + [ ] data model bug
    + [ ] tree utils bug
    + [x] function treelize bug (including method and classmethod)
    + [ ] code design/refactor
    + [ ] documentation request
    + [ ] new feature request
- [x] I have visited the [readme](https://github.com/opendilab/treevalue/blob/main/README.md) and [doc](https://opendilab.github.io/treevalue/main/index.html)
- [x] I have searched through the [issue tracker](https://github.com/opendilab/treevalue/issues) and [pr tracker](https://github.com/opendilab/treevalue/pulls)
- [x] I have mentioned version numbers, operating system and environment, where applicable:

```python
import sys

import treevalue

print(treevalue.__version__, sys.version, sys.platform)
```

Version 1.0.0, python 3.9.6 (should be independent of python's version)

The following code

python
```
from treevalue import func_treelize, FastTreeValue


@func_treelize(mode='outer', missing=lambda: [])
def append(arr: list, *args):
    print(arr, args)
    for item in args:
        if item:
            arr.append(item)
    return arr


t0 = FastTreeValue({})
t1 = FastTreeValue({'a': 2, 'x': {'c': 4, 'd': 9}})
t2 = FastTreeValue({'a': 4, 'b': 48, 'x': {'d': 54}})
t3 = FastTreeValue({'b': -12, 'x': 7, 'y': {'e': 3, 'f': 4}})

tr = append(t0, t1, t2, t3)
```

`treevalue graph`'s dumped graph is

![test_graph_10](https://user-images.githubusercontent.com/20508435/140470107-a8500fd1-3467-4535-9e43-93c89e244c04.png)

tr.x's value is obviously incorrect.

**Comment by HansBug** [comment]
The following code can also recurrence the bug

```python
t0 = FastTreeValue({})
t1 = FastTreeValue({'a': 2, 'b': 7, 'x': {'c': 4, 'd': 9}})
t2 = FastTreeValue({'a': 4, 'b': 48, 'x': {'c': -11, 'd': 54}})
t3 = FastTreeValue({'a': 9, 'b': -12, 'x': {'c': 3, 'd': 4}})

tr = append(t0, t1, t2, t3)
```

**Comment by HansBug** [comment]
Fixed in #11 .
